spec/crypto_env_var/cipher_spec.rb

```diff
@@ -41,12 +87,14 @@
       }.to raise_error OpenSSL::PKey::RSAError, /private key needed/
     end
 
+    include_examples "message integrity verification"
+
     it "can't decrypt a ciphertext generated with a different private key" do
-      other_private_key = OpenSSL::PKey::RSA.generate(2048)
-      other_cyphertext = other_private_key.private_encrypt(plaintext)
+      other_cipher = described_class.new(OpenSSL::PKey::RSA.generate(2048))
```

Comment:
Stupid question - how come we specificy the key size here, but not in the code?

Reply:
This is just to dynamically generate a throwaway key for this tests.
In a real-world scenario, the calling code is supposed to provide the keys as strings. Where they come from is up to the user.

The key size is not really important, as long as it's large enough to encrypt the random AES key and the SHA digest. If it's not, an error will be thrown asking the user to use a larger key.